framework/decode/decode_json_util.h

```diff
@@ -220,6 +220,56 @@ void FieldToJsonAsHex(nlohmann::ordered_json&                               jdat
     FieldToJsonAsHex(jdata, &data, options);
 }
 
+/// Same as array FieldToJson above but converts elements pointed-to to binary
```

Comment:
No reason I guess. It got on my todo list at some point so I did it. The binary looks nice for eight bit fields like stencil masks but I agree it is a bit meh for 32 bit ones that are all zero or 31 zeros and one one in the lowest bit (its not even a readability boon there as your eyes have to track along that row of zeros scanning for ones).

There is a P2 issue in the context of Vulkan toASCII that mentions this but over there we never resolved the choice between hex or binary: https://github.com/LunarG/gfxreconstruct/issues/768

I could break the binary out into a separate PR and make these hex. What do people want: a raw JSON decimal number (easy to parse), a string with hex value encoded in it, binary string, hex for long ones and binary for short? (if hex, fixed-length or variable length?)

Reply:
Ugly e.g.: 
```json
{"index":74,"method":{"name":"CreateHeap","thread":1,"object":{"type":"ID3D12Device","handle":5},"return":"S_OK","args":{"pDesc":{"SizeInBytes":24969216,"Properties":{"Type":"D3D12_HEAP_TYPE_DEFAULT","CPUPageProperty":"D3D12_CPU_PAGE_PROPERTY_UNKNOWN","MemoryPoolPreference":"D3D12_MEMORY_POOL_UNKNOWN","CreationNodeMask":"0b00000000000000000000000000000001","VisibleNodeMask":"0b00000000000000000000000000000001"},"Alignment":0,"Flags":"0x00000000"},"riid":"IID_ID3D12Heap","ppvHeap":3329}}}
```
Not ugly e.g.:
```json
{"index":315,"method":{"name":"CreateGraphicsPipelineState","thread":1,"object":{"type":"ID3D12Device","handle":6},"return":"S_OK","args":{"pDesc":{"pRootSignature":57,"VS":{"pShaderBytecode":"[Binary data]","BytecodeLength":3089},"PS":{"pShaderBytecode":"[Binary data]","BytecodeLength":4590},"DS":{"pShaderBytecode":null,"BytecodeLength":0},"HS":{"pShaderBytecode":null,"BytecodeLength":0},"GS":{"pShaderBytecode":null,"BytecodeLength":0},"StreamOutput":{"pSODeclaration":null,"NumEntries":0,"pBufferStrides":null,"NumStrides":0,"RasterizedStream":0},"BlendState":{"AlphaToCoverageEnable":false,"IndependentBlendEnable":false,"RenderTarget":[{"BlendEnable":false,"LogicOpEnable":false,"SrcBlend":"D3D12_BLEND_SRC_ALPHA","DestBlend":"D3D12_BLEND_INV_SRC_ALPHA","BlendOp":"D3D12_BLEND_OP_ADD","SrcBlendAlpha":"D3D12_BLEND_ONE","DestBlendAlpha":"D3D12_BLEND_INV_SRC_ALPHA","BlendOpAlpha":"D3D12_BLEND_OP_ADD","LogicOp":"D3D12_LOGIC_OP_CLEAR","RenderTargetWriteMask":"0b00001111"},{"BlendEnable":false,"LogicOpEnable":false,"SrcBlend":"Unhandled D3D12_BLEND","DestBlend":"Unhandled D3D12_BLEND","BlendOp":"Unhandled D3D12_BLEND_OP","SrcBlendAlpha":"Unhandled D3D12_BLEND","DestBlendAlpha":"Unhandled D3D12_BLEND","BlendOpAlpha":"Unhandled D3D12_BLEND_OP","LogicOp":"D3D12_LOGIC_OP_CLEAR","RenderTargetWriteMask":"0b00000000"},{"BlendEnable":false,"LogicOpEnable":false,"SrcBlend":"Unhandled D3D12_BLEND","DestBlend":"Unhandled D3D12_BLEND","BlendOp":"Unhandled D3D12_BLEND_OP","SrcBlendAlpha":"Unhandled D3D12_BLEND","DestBlendAlpha":"Unhandled D3D12_BLEND","BlendOpAlpha":"Unhandled D3D12_BLEND_OP","LogicOp":"D3D12_LOGIC_OP_CLEAR","RenderTargetWriteMask":"0b00000000"},{"BlendEnable":false,"LogicOpEnable":false,"SrcBlend":"Unhandled D3D12_BLEND","DestBlend":"Unhandled D3D12_BLEND","BlendOp":"Unhandled D3D12_BLEND_OP","SrcBlendAlpha":"Unhandled D3D12_BLEND","DestBlendAlpha":"Unhandled D3D12_BLEND","BlendOpAlpha":"Unhandled D3D12_BLEND_OP","LogicOp":"D3D12_LOGIC_OP_CLEAR","RenderTargetWriteMask":"0b00000000"},{"BlendEnable":false,"LogicOpEnable":false,"SrcBlend":"Unhandled D3D12_BLEND","DestBlend":"Unhandled D3D12_BLEND","BlendOp":"Unhandled D3D12_BLEND_OP","SrcBlendAlpha":"Unhandled D3D12_BLEND","DestBlendAlpha":"Unhandled D3D12_BLEND","BlendOpAlpha":"Unhandled D3D12_BLEND_OP","LogicOp":"D3D12_LOGIC_OP_CLEAR","RenderTargetWriteMask":"0b00000000"},{"BlendEnable":false,"LogicOpEnable":false,"SrcBlend":"Unhandled D3D12_BLEND","DestBlend":"Unhandled D3D12_BLEND","BlendOp":"Unhandled D3D12_BLEND_OP","SrcBlendAlpha":"Unhandled D3D12_BLEND","DestBlendAlpha":"Unhandled D3D12_BLEND","BlendOpAlpha":"Unhandled D3D12_BLEND_OP","LogicOp":"D3D12_LOGIC_OP_CLEAR","RenderTargetWriteMask":"0b00000000"},{"BlendEnable":false,"LogicOpEnable":false,"SrcBlend":"Unhandled D3D12_BLEND","DestBlend":"Unhandled D3D12_BLEND","BlendOp":"Unhandled D3D12_BLEND_OP","SrcBlendAlpha":"Unhandled D3D12_BLEND","DestBlendAlpha":"Unhandled D3D12_BLEND","BlendOpAlpha":"Unhandled D3D12_BLEND_OP","LogicOp":"D3D12_LOGIC_OP_CLEAR","RenderTargetWriteMask":"0b00000000"},{"BlendEnable":false,"LogicOpEnable":false,"SrcBlend":"Unhandled D3D12_BLEND","DestBlend":"Unhandled D3D12_BLEND","BlendOp":"Unhandled D3D12_BLEND_OP","SrcBlendAlpha":"Unhandled D3D12_BLEND","DestBlendAlpha":"Unhandled D3D12_BLEND","BlendOpAlpha":"Unhandled D3D12_BLEND_OP","LogicOp":"D3D12_LOGIC_OP_CLEAR","RenderTargetWriteMask":"0b00000000"}]},"SampleMask":"0b11111111111111111111111111111111","RasterizerState":{"FillMode":"D3D12_FILL_MODE_SOLID","CullMode":"D3D12_CULL_MODE_NONE","FrontCounterClockwise":true,"DepthBias":0,"DepthBiasClamp":0.0,"SlopeScaledDepthBias":0.0,"DepthClipEnable":true,"MultisampleEnable":false,"AntialiasedLineEnable":false,"ForcedSampleCount":0,"ConservativeRaster":"D3D12_CONSERVATIVE_RASTERIZATION_MODE_OFF"},"DepthStencilState":{"DepthEnable":false,"DepthWriteMask":"D3D12_DEPTH_WRITE_MASK_ZERO","DepthFunc":"D3D12_COMPARISON_FUNC_ALWAYS","StencilEnable":false,"StencilReadMask":"0b11111111","StencilWriteMask":"0b11111111","FrontFace":{"StencilFailOp":"D3D12_STENCIL_OP_KEEP","StencilDepthFailOp":"D3D12_STENCIL_OP_KEEP","StencilPassOp":"D3D12_STENCIL_OP_KEEP","StencilFunc":"D3D12_COMPARISON_FUNC_ALWAYS"},"BackFace":{"StencilFailOp":"D3D12_STENCIL_OP_KEEP","StencilDepthFailOp":"D3D12_STENCIL_OP_KEEP","StencilPassOp":"D3D12_STENCIL_OP_KEEP","StencilFunc":"D3D12_COMPARISON_FUNC_ALWAYS"}},"InputLayout":{"pInputElementDescs":null,"NumElements":0},"IBStripCutValue":"D3D12_INDEX_BUFFER_STRIP_CUT_VALUE_DISABLED","PrimitiveTopologyType":"D3D12_PRIMITIVE_TOPOLOGY_TYPE_TRIANGLE","NumRenderTargets":1,"RTVFormats":["DXGI_FORMAT_R10G10B10A2_UNORM","DXGI_FORMAT_UNKNOWN","DXGI_FORMAT_UNKNOWN","DXGI_FORMAT_UNKNOWN","DXGI_FORMAT_UNKNOWN","DXGI_FORMAT_UNKNOWN","DXGI_FORMAT_UNKNOWN","DXGI_FORMAT_UNKNOWN"],"DSVFormat":"DXGI_FORMAT_UNKNOWN","SampleDesc":{"Count":1,"Quality":0},"NodeMask":"0b00000000000000000000000000000001","CachedPSO":{"pCachedBlob":null,"CachedBlobSizeInBytes":0},"Flags":"0x00000000"},"riid":"IID_ID3D12PipelineState","ppPipelineState":61}}}
```

Reply:
I can't convert hex to binary easily in my head past a byte or so.  In general any value works because I imagine a tool has to convert the number back to the masks or bitfield or something in order to say something useful.  Maybe file an issue suggesting it should be hex but I'm not interested in forcing Andrew to do this in his remaining day.

Reply:
#1377 created.